### ISSUE TITLE
<details> breaks nested Datatable

### ISSUE THREAD

**Issue by HillbillyBishop** [wet-boew/wet-boew#8612]
<details id="structurePanel">
        <summary>Structure</summary>
        <table class="display mrgn-tp-md mrgn-bttm-md" id="structureTable">
                        <thead>
                            <tr class="headerRow">
                                <th hidden></th>
                                <th>@Resources.Label.MOSID</th>
                                <th>@Resources.Label.Name</th>
                                <th>@Resources.Label.Component</th>
                                <th>@Resources.Label.SubComponent</th>
                            </tr>
                        </thead>
                        <tbody>
                            @foreach (var subdiv in Model.qualification_instance.GetSubDivisions())
                            {
                                <tr>
                                    <td hidden>@subdiv.SubDivInst_ID</td>
                                    <td>@subdiv.GetDetailedMOSID()</td>
                                    <td>@subdiv.getMOSIDTitle("en")</td>
                                    <td>@subdiv.getComponentTitle("en")</td>
                                    <td>@subdiv.getSubCompTitle("en")</td>
                                </tr>
                            }
                        </tbody>
          </table>
</details>

The above code builds without error, but the produced datatable: $("#structureTable").DataTable();
does not allow sorting, filtering, pagination, or any other scripted actions to occur, it simply populates & formats as datatable. The bug occurs silently.
If the table is moved outside <details>, it functions correctly.

**Comment by thekodester** [comment]
Using `$("#structureTable").DataTable();` is unnecessary in WET. Please review https://wet-boew.github.io/v4.0-ci/demos/tables/tables-en.html and make sure to use the classes `wb-tables table`.

**Comment by HillbillyBishop** [comment]
I am implementing datatable extensions (rowReorder, Select, and Scroller), ex:  
``` 
$("#structureTable").DataTable()
        select: true,
        rowReorder: true,
        columnDefs: [
            { targets:'_all', visible: true }
        ]
    });
```
and am unable to add `data-wb-tables='{ rowReorder:true }'`. Are datatable extensions supported, or am I just using the incorrect syntax?

**Comment by thekodester** [comment]
`rowReorder` requires an additional JS and CSS file. You can see here: https://datatables.net/extensions/rowreorder/examples/initialisation/simple.html.

```
<link rel="stylesheet" href="https://cdn.datatables.net/rowreorder/1.2.5/css/rowReorder.dataTables.min.css" />
<script src="https://cdn.datatables.net/rowreorder/1.2.5/js/dataTables.rowReorder.min.js"></script>
```

They're two ways you can go about it. One is the what you are currently doing and not using `wb-tables table`. You well need to also manually link all other need DataTable css and js file (since not using WET it won't preload). Cons to this approach are you well lose WET accessibility added to DataTables.

Another way you can do this it not the best thing, but works is:

Included:

`<link rel="stylesheet" href="https://cdn.datatables.net/rowreorder/1.2.5/css/rowReorder.dataTables.min.css" />`

in your `<head></head>`.

Included:

```
<script>
wb.doc.on( "wb-ready.wb-tables", function() {
Modernizr.load( {
load: [
"https://cdn.datatables.net/rowreorder/1.2.5/js/dataTables.rowReorder" + wb.getMode() + ".js",
],
complete: function() {
$( "#rowReorderTable" ).dataTable().fnDestroy();
$( "#rowReorderTable" ).DataTable( {
rowReorder: true
} );
}
} );
} );
</script>
```

Under your scripts at the end `<body></body>` or in a separate file.

**Comment by thekodester** [comment]
This is basically the same for anything that requires additional JS or CSS.

**Comment by HillbillyBishop** [comment]
Solution 1 is what I've been trying, all scripts & css are loaded (the datatable + extensions work anywhere on the page _except_ for inside <details> block).

Solution 2 I will try now.

Thanks so much for assistance.

**Comment by thekodester** [comment]
Yeah neither are prefect cause, partly due to the way WET is designed to load the JS file using `Modernizr.load(...)`.

Also as for another question you had, no it not supported out of the box, since WET would need to load a ridiculous amount of additional JS and CSS files that may not be used.

**Comment by thekodester** [comment]
Just wondering if you got it working with the other method?
